library/kani/src/lib.rs

```diff
@@ -74,38 +75,31 @@ pub fn any<T: Arbitrary>() -> T {
 
 /// This function creates an unconstrained value of type `T`. This may result in an invalid value.
 ///
-/// # Example:
-///
-/// In the snippet below, we are verifying the behavior of the function `fn_under_verification`
-/// under all possible values of char, including invalid ones that are greater than char::MAX.
-///
-/// ```rust
-/// let inputA = unsafe { kani::any_raw::<char>() };
-/// fn_under_verification(inputA);
-/// ```
-///
 /// # Safety
 ///
 /// This function is unsafe and it may represent invalid `T` values which can lead to many
-/// undesirable undefined behaviors. Users must validate that the symbolic variable respects
-/// the type invariant as well as any other constraint relevant to their usage. E.g.:
+/// undesirable undefined behaviors. Users must guarantee that an unconstrained symbolic value
+/// for type T only represents valid values.
 ///
-/// ```rust
-/// let c = unsafe { kani::any_raw::char() };
-/// kani::assume(char::from_u32(c as u32).is_ok());
-/// ```
+/// # Deprecation
 ///
-#[rustc_diagnostic_item = "KaniAnyRaw"]
+/// We have decided to deprecate this function due to the fact that its result can be the source
+/// of undefined behavior.
 #[inline(never)]
+#[deprecated(
+    since = "0.8.0",
+    note = "This function may return symbolic values that don't respects the language type invariants."
+)]
 pub unsafe fn any_raw<T>() -> T {
```

Comment:
Should we also `doc_hidden` ?

Reply:
Sure. I don't think we publish our crates doc yet, but hopefully soon.